lib/WasmReader/BaseWasmReader.h

```diff
@@ -8,19 +8,36 @@
 namespace Wasm
 {
 
-class BaseWasmReader
-{
-public:
-    virtual WasmOp ReadFromModule() = 0;
-    virtual WasmOp ReadExpr() = 0;
-    virtual WasmOp ReadFromBlock() = 0;
-    virtual WasmOp ReadFromCall() = 0;
-    virtual bool IsBinaryReader() = 0;
-    WasmNode    m_currentNode;
-    ModuleInfo * m_moduleInfo;
-
-protected:
-    WasmFunctionInfo *  m_funcInfo;
-};
+    enum SectionFlag
+    {
+        fSectNone,
+        fSectIgnore,
```

Comment:
I didn't know at this time, I just figure it will make it easy to extend if need be.
If we realize we don't need it (ie: when all sections are read) then we could drop it